packages/themes/scss/_mixins.scss

```diff
@@ -1,170 +1,35 @@
-    --field-02: #{map-get($theme, 'field-02')};
-    --inverse-01: #{map-get($theme, 'inverse-01')};
-    --inverse-02: #{map-get($theme, 'inverse-02')};
-    --support-01: #{map-get($theme, 'support-01')};
-    --support-02: #{map-get($theme, 'support-02')};
-    --support-03: #{map-get($theme, 'support-03')};
-    --support-04: #{map-get($theme, 'support-04')};
-    --inverse-support-01: #{map-get($theme, 'inverse-support-01')};
-    --inverse-support-02: #{map-get($theme, 'inverse-support-02')};
-    --inverse-support-03: #{map-get($theme, 'inverse-support-03')};
-    --inverse-support-04: #{map-get($theme, 'inverse-support-04')};
-    --overlay-01: #{map-get($theme, 'overlay-01')};
-    --danger: #{map-get($theme, 'danger')};
-    --focus: #{map-get($theme, 'focus')};
-    --inverse-focus-ui: #{map-get($theme, 'inverse-focus-ui')};
-    --hover-primary: #{map-get($theme, 'hover-primary')};
-    --active-primary: #{map-get($theme, 'active-primary')};
-    --hover-primary-text: #{map-get($theme, 'hover-primary-text')};
-    --hover-secondary: #{map-get($theme, 'hover-secondary')};
-    --active-secondary: #{map-get($theme, 'active-secondary')};
-    --hover-tertiary: #{map-get($theme, 'hover-tertiary')};
-    --active-tertiary: #{map-get($theme, 'active-tertiary')};
-    --hover-ui: #{map-get($theme, 'hover-ui')};
-    --active-ui: #{map-get($theme, 'active-ui')};
-    --selected-ui: #{map-get($theme, 'selected-ui')};
-    --hover-selected-ui: #{map-get($theme, 'hover-selected-ui')};
-    --inverse-hover-ui: #{map-get($theme, 'inverse-hover-ui')};
-    --hover-danger: #{map-get($theme, 'hover-danger')};
-    --active-danger: #{map-get($theme, 'active-danger')};
-    --hover-row: #{map-get($theme, 'hover-row')};
-    --visited-link: #{map-get($theme, 'visited-link')};
-    --disabled-01: #{map-get($theme, 'disabled-01')};
-    --disabled-02: #{map-get($theme, 'disabled-02')};
-    --disabled-03: #{map-get($theme, 'disabled-03')};
-    --highlight: #{map-get($theme, 'highlight')};
-    --skeleton-01: #{map-get($theme, 'skeleton-01')};
-    --skeleton-02: #{map-get($theme, 'skeleton-02')};
-    --brand-01: #{map-get($theme, 'brand-01')};
-    --brand-02: #{map-get($theme, 'brand-02')};
-    --brand-03: #{map-get($theme, 'brand-03')};
-    --active-01: #{map-get($theme, 'active-01')};
-    --hover-field: #{map-get($theme, 'hover-field')};
+/// @access private
+/// @group @carbon/themes
+@mixin custom-property($name, $value, $prefix: $custom-property-prefix) {
+  @if type-of($value) == map {
+    @each $property, $property-value in $value {
+      @include custom-property('#{$name}-#{$property}', $property-value);
+    }
+  } @else {
+    --#{$prefix}-#{$name}: #{$value};
   }
+}
 
-  @content;
-  // Reset to default theme after apply in content
-  @if $theme != $carbon--theme {
-    @include carbon--theme();
+/// @access private
+/// @group @carbon/themes
+@function should-emit($theme-a, $theme-b, $token, $emit-difference) {
```

Comment:
Can this go in a `_functions.scss` file?

Reply:
Same notes as above, maybe this could be a broader topic change? Thankfully we could swap any of this out in the future so shouldn't have to block anything now.

Reply:
Ah, it's designated as private. I don't have a preference on where private functions live. Feel free to resolve this convo.